src/Application/ValueCounter.php

```diff
@@ -4,10 +4,8 @@
 
 namespace ProfessionalWiki\WikibaseFacetedSearch\Application;
 
-use Wikibase\DataModel\Entity\PropertyId;
-
 interface ValueCounter {
 
-	public function countValues( PropertyId $property ): ValueCounts;
+	public function countValues( PropertyConstraints $constraints ): ValueCounts;
```

Comment:
I guess this slightly changes the semantics of the method (i.e. counting the _state_, instead of a _property_), but PropertyConstraints contains PropertyId, so it seemed silly to include both.